repose-aggregator/components/filters/valkyrie-authorization/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala

```diff
@@ -180,6 +220,76 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
     }
   }
 
+
+  def cullResponse(url: String, response: MutableHttpServletResponse, devicePermissions: DeviceList): Unit = {
+
+    def getJsPathFromString(jsonPath: String): JsPath = {
+      val pathTokens: List[PathToken] = JSONPath.parser.compile(jsonPath).getOrElse({
+        throw new ResponseCullingException(s"Unable to parse JsonPath: $jsonPath")
+      })
+      pathTokens.foldLeft(new JsPath) { (path, token) =>
+        token match {
+          case RootNode => path
+          case Field(name) => path \ name
+        }
+      }
+    }
+
+    val matchingResources: mutable.Buffer[Resource] = Option(configuration.getCollectionResources)
+      .getOrElse(new CollectionResources)
+      .getResource.asScala.filter(_.getPathRegex.r.findFirstMatchIn(url).isDefined)
+    if (matchingResources.nonEmpty) {
```

Comment:
Also, I'm not sure if the code can be broken down more than this, but it would be great if so.